contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+     *
+     * This function has a slightly different behaviours depending on the status of the underlying Aerodrome
+     * slipstream pool. The function achieves its behaviour using the following 3 steps:
+     * 1. withdrawPartialLiqidity -> so that moving the activeTrading price via  a swap is cheaper
+     * 2. swapToDesiredPosition   -> move active trading price in the pool to be able to deposit WETH & OETHb
+     *                               tokens with the desired pre-configured shares
+     * 3. addLiquidity            -> add liquidity into the pool respecting share split configuration
+     *
+     * Scenario 1: When there is yet no liquidity yet in the pool (from the strategy or others) that means that
+     *             someone has minted the pool, added the initial liquidity and removed the liquidity from the
+     *             pool. (See `aerodrome_amo_liquidity.py` brownie script on block 18558804 to test the situation).
+     *             Then the Aerodrome pool is in this particular state where active trading price of the pool is
+     *             still at the value when last liquidity was in the pool and that trading price can not be moved
+     *             since there is no liquidity to perform the swap. In such a case the rebalancing transaction
+     *             shall be reverted.
+     *             Swap transaction is the one that shall fail. Unfortunately there is no easy way to query Aerodrome
+     *             slipstream pool for the amount of liquidity deposited. Even tokens on the contract can be from
+     *             fees, or liquidity that has been removed but not yet claimed.
+     *             It becomes the responsibility of the strategist or deployer to add some liquidity in the configured
+     *             tick ranges to the pool to facilitate the swap. Effectively turning Scenario 1 into a Scenario 2
+     * Scenario 2: When there is no liquidity in the pool from the strategy but there is from other LPs then
+     *             only step 1 is skipped. (It is important to note that liquidity needs to exist in the configured
+     *             strategy tick ranges in order for the swap to be possible) Step 3 mints new liquidity position
+     *             instead of adding to an existing one.
+     * Scenario 3: When there is strategy's liquidity in the pool all 3 steps are taken
+     *
+     *
+     * Exact _amountToSwap, _minTokenReceived & _swapWeth parameters shall be determined by simulating the
+     * transaction off-chain. The strategy checks that after the swap the share of the tokens is in the
+     * expected ranges.
+     *
+     * @param _amountToSwap The amount of the token to swap
+     * @param _minTokenReceived Slippage check -> minimum amount of token expected in return
+     * @param _swapWeth Swap using WETH when true, use OETHb when false
+     */
+    function rebalance(
+        uint256 _amountToSwap,
+        uint256 _minTokenReceived,
```

Comment:
Perhaps `_minTokenReceived` should go to the end, since that's the normal place for it, and `_swapWeth` should be next to `_amountToSwap` since that's what it changes.

Reply:
agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/71ef9511e2356d9d6d5d2d20688d6506fe48da99